src/lib/__tests__/AnnotationControlsFSM-test.js

```diff
@@ -1,86 +1,134 @@
+    describe('AnnotationState.NONE', () => {
+        [
+            {
+                input: AnnotationInput.CLICK,
+                mode: AnnotationMode.HIGHLIGHT,
+                nextState: AnnotationState.HIGHLIGHT,
+                output: AnnotationMode.HIGHLIGHT,
+            },
+            {
+                input: AnnotationInput.CLICK,
+                mode: AnnotationMode.REGION,
+                nextState: AnnotationState.REGION,
+                output: AnnotationMode.REGION,
+            },
+            {
+                input: AnnotationInput.CREATE,
+                mode: AnnotationMode.HIGHLIGHT,
+                nextState: AnnotationState.HIGHLIGHT_TEMP,
+                output: AnnotationMode.HIGHLIGHT,
+            },
+            {
+                input: AnnotationInput.CREATE,
+                mode: AnnotationMode.REGION,
+                nextState: AnnotationState.REGION_TEMP,
+                output: AnnotationMode.REGION,
+            },
+        ].forEach(({ input, mode, nextState, output }) => {
+            it(`should go to state ${nextState} and output ${output}`, () => {
+                const annotationControlsFSM = new AnnotationControlsFSM();
+
+                expect(annotationControlsFSM.transition(input, mode)).to.equal(output);
+                expect(annotationControlsFSM.getState()).to.equal(nextState);
+            });
+        });
+
+        [AnnotationInput.CANCEL, AnnotationInput.SUCCESS, AnnotationInput.UPDATE].forEach(input => {
+            it(`should stay in state none if input is ${input}`, () => {
+                const annotationControlsFSM = new AnnotationControlsFSM();
+
+                expect(annotationControlsFSM.transition(input)).to.equal(AnnotationMode.NONE);
+                expect(annotationControlsFSM.getState()).to.equal(AnnotationState.NONE);
+            });
+        });
+    });
+
+    describe('AnnotationState.HIGHLIGHT/REGION', () => {
+        [AnnotationState.HIGHLIGHT, AnnotationState.REGION].forEach(state => {
+            // non-Click input
+            [AnnotationInput.CANCEL, AnnotationInput.CREATE, AnnotationInput.SUCCESS, AnnotationInput.SUCCESS].forEach(
+                input => {
+                    it(`should stay in state ${state}`, () => {
+                        const annotationControlsFSM = new AnnotationControlsFSM(state);
+
+                        expect(annotationControlsFSM.transition(input)).to.equal(state);
+                        expect(annotationControlsFSM.getState()).to.equal(state);
+                    });
+                },
+            );
+
+            // Click input
+            [AnnotationMode.HIGHLIGHT, AnnotationMode.REGION].forEach(mode => {
+                it(`should go to state none/${mode} if input is click and mode is ${mode}`, () => {
+                    const annotationControlsFSM = new AnnotationControlsFSM(state);
+
+                    const output = state === mode ? AnnotationMode.NONE : mode;
```

Comment:
We should avoid conditional logic in the tests, could we include it in the input parameters for the test?